annif/project.py

```diff
@@ -117,6 +130,16 @@ def analyzer(self):
                     project_id=self.project_id)
         return self._analyzer
 
+    @property
+    def transformer(self):
+        if self._transformer is None:
+            if self.transformer_spec:
+                self._transformer = annif.transformer.get_transformer(
+                    self.transformer_spec, project=self)
+            else:
+                self._transformer = None
```

Comment:
Assigning None here is basically redundant (`self._transformer` was already None) and it means that the decision not to use an analyzer is never cached. Maybe that's no big deal since the cost of redoing is just an additional if check for the spec. Anyway I suggest dropping the `else` block as it's not doing anything useful.

Reply:
Another option would be to assign a passthrough transformer here in the `else` block when no spec is set. That could actually be wise, because then `self.transformer` could never return None.